x/marker/keeper/keeper.go

```diff
@@ -267,22 +268,30 @@ func (k Keeper) GetSendDenyList(ctx sdk.Context, markerAddr sdk.AccAddress) []sd
 
 // AddSetNetAssetValues adds a set of net asset values to a marker
 func (k Keeper) AddSetNetAssetValues(ctx sdk.Context, marker types.MarkerAccountI, netAssetValues []types.NetAssetValue, source string) error {
+	var errs []error
 	for _, nav := range netAssetValues {
 		if nav.Price.Denom == marker.GetDenom() {
-			return fmt.Errorf("net asset value denom cannot match marker denom %q", marker.GetDenom())
+			errs = append(errs, fmt.Errorf("net asset value denom cannot match marker denom %q", marker.GetDenom()))
+			continue
 		}
+
 		if nav.Price.Denom != types.UsdDenom {
 			_, err := k.GetMarkerByDenom(ctx, nav.Price.Denom)
 			if err != nil {
-				return fmt.Errorf("net asset value denom does not exist: %v", err.Error())
+				if err2 := nav.Validate(); err2 == nil {
+					navEvent := types.NewEventSetNetAssetValue(marker.GetDenom(), nav.Price, nav.Volume, source)
```

Comment:
Why are we emitting an event for a net asset value that is being skipped?

Reply:
less events are better based on recent understanding...

Reply:
We want a record of the NAV data somewhere, even if it doesn't get recorded in state. This is a reflection of `SetNetAssetValue` which always emits the event as long as the NAV is valid. It too might emit this event without the NAV being written to state.

This `AddSetNetAssetValues` method was designed for the `AddNetAssetValues` endpoint. I don't know enough about the business requirements on that to properly try to remove some of the failure points, though. So, even though I feel like we should record NAVS even if there's no marker for the price denom, I wasn't comfortable removing this failure point. But at least this way, the data still goes somewhere.

In the exchange module, where `AddSetNetAssetValues` is also called, there's a bunch of other stuff that also helps ensures this event is emitted, even if it can't be stored to state.

Reply:
Emitting events isn't really important with the new state streaming direction that the Cosmos SDK is going ... eventually we will may remove all events from the block results.